prompter-source.lisp

```diff
@@ -492,12 +454,25 @@ Suggestions are made with the `suggestion-maker' slot from `source'."))
 
 (defmethod attribute-key ((attribute t))
   (first attribute))
-(defmethod attribute-value ((attribute t))
-  (second attribute))
+
+(export-always 'attribute-value)
+(define-generic attribute-value ((attribute t) &key wait-p)
+  "Return value of ATTRIBUTE.
+If WAIT-P, block until attribute is computed.
+Otherwise return a `lparallel:future' it the attribute is not done calculating."
+  (if (or wait-p
```

Comment:
I'd perhaps rename "wait-p" to "synchronous-p", but it all comes down to personal preference.

If you it beneficial to rename it, notice that it requires query-replacing it.

Reply:
"wait" is more idiomatic, as used by Bordeaux-Threads and others.
Also the intent of `wait` is clear, while the one of `synchronous` is less so at first.